transcom/mymove
# B 18109 lowest weight

## [B-18109](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A871306&RoomContext=TeamRoom%3A808731&concept=TeamRoom)

## Summary

- [x] Frontend properly calculates weights of diverted shipments, no longer "double counting" the weight of a diverted shipment
- [x] Reweigh creation supports diversions
- [x] Reweigh updating supports diversions
- [ ] Create shipment supports diversions (By inheriting parent shipment's `primeActualWeight`

This ticket and PR provides the following 

## Prime API enhancements:
- On reweigh creation or update, if the reweigh is for a diverted shipment the following actions occur:
  - If this new reweigh is a lower weight than all reweighs found in the chain
    - It will go through as a new reweigh and also subsequently update all existing reweighs in the "diversion chain" to now hold this new lowest weight value.
  - If this new reweigh is a higher weight than any reweigh found in the chain
    - It will be rejected. Per customer requirements we are to only allow the lowest weights for a diverted shipment
- On shipment creation, if the shipment is a diversion the following actions occur:
  - New shipment automatically inherits the parent shipment's `primeActualWeight` value to be used in payment requests from the prime (Should they not provide a weight billed for a partial payment)

## Frontend enhancements:
- Diverted shipments are now chained together by their addresses as well as the `diversion` boolean on the frontend. This allows us to identify which shipments are diversions and identify the proper parent/child chain. After we get this chain we can grab the lowest weight and report that.
  - This resolves "estimated" and "net" weights being double counted for diversions and now also reflects the new rule of only using the lowest weight.

## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Update Postman to utilize the latest Prime V2 endpoints (Make sure your environment variables in postman point to the correct V2 endpoint)
2. Login as a
3.

## Screenshots
